common/persistence/versionHistory.go

```diff
@@ -21,174 +21,517 @@
 package persistence
 
 import (
+	"bytes"
 	"fmt"
+
 	"github.com/uber/cadence/.gen/go/shared"
 	"github.com/uber/cadence/common"
-	"reflect"
 )
 
-// NewVersionHistory initializes new version history
-func NewVersionHistory(items []VersionHistoryItem) VersionHistory {
-	if len(items) == 0 {
-		panic("version history items cannot be empty")
+// NewVersionHistoryItem create a new version history item
+func NewVersionHistoryItem(
+	inputEventID int64,
+	inputVersion int64,
+) *VersionHistoryItem {
+
+	if inputEventID < 0 || inputVersion < 0 {
+		panic(fmt.Sprintf(
```

Comment:
You have plans to change this and other panics to return error in future, right?

Reply:
i need to think about this.

based on my current assumption
event ID < 0 is bug, should fail fast
version == common.EmptyVersion ( < 0 ) can happen for local workflow, however, this is not supported yet